solidity-plugin/src/main/java/org/sonarsource/solidity/UtilsSensor.java

```diff
@@ -0,0 +1,98 @@
+package org.sonarsource.solidity;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.util.Optional;
+import org.antlr.v4.runtime.ParserRuleContext;
+import org.antlr.v4.runtime.Token;
+import org.antlr.v4.runtime.tree.ParseTree;
+import org.sonar.api.batch.fs.InputFile;
+import org.sonar.api.utils.log.Logger;
+import org.sonar.api.utils.log.Loggers;
+import org.sonarsource.solidity.frontend.SolidityParser;
+import org.sonarsource.solidity.frontend.SolidityParser.ExpressionContext;
+import org.sonarsource.solidity.frontend.SolidityParser.IfStatementContext;
+import org.sonarsource.solidity.frontend.SolidityParser.ReturnStatementContext;
+import org.sonarsource.solidity.frontend.SolidityParser.SimpleStatementContext;
+import org.sonarsource.solidity.frontend.SolidityParser.StatementContext;
+import org.sonarsource.solidity.frontend.SolidityParser.VariableDeclarationStatementContext;
+
+public final class UtilsSensor {
+
+  private static final Logger LOG = Loggers.get(UtilsSensor.class);
+
+  public static Optional<ParseTree> checkForElseStatement(ParserRuleContext ctxNode) {
+    if (!ctxNode.children.isEmpty() && ctxNode.children.size() >= 6) {
+      // the 6th child is where else exists even for else-if case
+      Token token = (Token) ctxNode.children.get(5).getPayload();
+      if (token.getType() == 40) {
+        ParseTree child6 = ctxNode.children.get(6);
+        // exclude else - if cases
+        if (!child6.getChild(0).getClass().equals(IfStatementContext.class))
+          return Optional.of(child6);
+      }
+    }
+    return Optional.empty();
+  }
+
+  public static boolean isElseIfStatement(IfStatementContext ctx) {
+    return ctx.getParent().getRuleIndex() == 38 && ctx.getParent().getParent().getRuleIndex() == 40;
+  }
+
+  public static int emptyLines(InputFile file) {
+    String line;
+    int emptyLinesCounter = 0;
+    try {
+      BufferedReader reader = new BufferedReader(new InputStreamReader(file.inputStream()));
+      while ((line = reader.readLine()) != null) {
+        if (line.trim().length() == 0) {
+          emptyLinesCounter++;
+        }
+      }
+    } catch (IOException e) {
+      LOG.debug(e.getMessage(), e);
+    }
+    // +1 for the last line
+    return emptyLinesCounter;
+  }
+
+  public static boolean isTernaryExpression(StatementContext ctx) {
+    // TODO
```

Comment:
![Code Smell](https://sonarsource.github.io/sonar-github/code_smell.png 'Code Smell') Code Smell: Complete the task associated to this TODO comment. ([squid:S1135](https://sonarcloud.io/organizations/stylianos-agapiou-sonarsource-gi/rules#rule_key=squid%3AS1135))

[See it in SonarCloud](https://sonarcloud.io/project/issues?id=org.sonarsource.solidity%3Asonar-solidity&pullRequest=23&issues=AWQ2bn0X_bEzQ-kUlT1Z&open=AWQ2bn0X_bEzQ-kUlT1Z)

[ISSUE_KEY]: AWQ2bn0X_bEzQ-kUlT1Z